.github/workflows/release.yml

```diff
@@ -31,14 +31,26 @@ jobs:
         uses: actions/setup-go@d35c59abb061a4a6fb18e82ac0862c26744d6ab5 # pin@5.5.0
         with:
           go-version: '${{ env.GOLANG_VERSION }}'
+
       - name: Run unit tests
         run: make test-coverage
+
       - name: Build Helm Binaries
         run: |
           set -eu -o pipefail
 
-          make build-cross VERSION="${{ github.ref_name }}"
-          make dist checksum VERSION="${{ github.ref_name }}"
+      - name: Run Go Release
+        uses: goreleaser/goreleaser-action@v6.3.0
+        if: success() && startsWith(github.ref, 'refs/tags/')
+        with:
+          version: latest
+          args: release --rm-dist
+        env:
+          GITHUB_TOKEN: ${{ secrets.GH_PAT }}
+
+      - name: Verify checksum
+        run: |
+            make dist checksum VERSION="${{ github.ref_name }}"
```

Comment:
unless I'm mistaken, `make dist` is recreating the release archives. Which `.goreleaser.yaml` is also configured to do:
https://github.com/helm/helm/pull/30913/files#diff-413d66267886290506ba7e2df00c65cb65341ec7347cd34568117ff435f63623R25

please use one or the other (preference would be to use goreleaser, rather than makefile logic I think)